src/cli/lint/index.ts

```diff
@@ -13,7 +13,7 @@ export const lint = async (
   tscOutputStream: Writable | undefined = undefined,
   workerThreads = true,
 ) => {
-  await Promise.all([tryRefreshIgnoreFiles(), upgradeSkuba()]);
+  await Promise.all([tryRefreshIgnoreFiles({ linting: true }), upgradeSkuba()]);
```

Comment:
Sorry for the late review, here are my thoughts:

1. Totally agree that we need to be careful about automatically committing `.npmrc` changes
2. I'd definitely want to retain ignore file autofixes on the happy path. Another option may be to revisit whether we should fail the command if autofix fails, or perhaps if autofix fails _and_ includes a refreshed ignore file.

Reply:
There's a chance that this could tie in to https://github.com/seek-oss/skuba/pull/1274#discussion_r1424898271; maybe patches should declare whether they are mandatory, in which case they should fail the build if not autofixed?